### PR TITLE
closes #750 - support account-level logpush jobs

### LINKED ISSUE #750: support account-level logpush jobs
### Current cloudflare-go version

latest

### Description

Logpush supports both zone-level and account-level logpush jobs, however the provider currently requires zone_id arguments for both both cloudflare_logpush_job and cloudflare_logpush_ownership_challenge resources. Account-level jobs therefore can't be created in Terraform. If the provider accepted either a zone ID or an account ID, it could create either type of job.

The API supports both, but it appears cloudflare-go only supports zone IDs as well.
https://github.com/cloudflare/cloudflare-go/blob/master/logpush.go#L98-L99

Docs link:
https://developers.cloudflare.com/logs/reference/log-fields

### Use cases

required to support this feature in the terraform provider to manage logpush jobs for:
* Audit Logs
* Gateway DNS
* Gateway HTTP
* Gateway Network (when that lands)

### Potential cloudflare-go usage
To keep API compatibility, existing functions could add a new `scope` or `IDtype` or similar parameter to determine if the ID being passed in is a zone or account scope.  It could default to `zone`
```go
func (api *API) CreateLogpushJob(ctx context.Context, ID string, job LogpushJob, IDtype string) (*LogpushJob, error) {
    APIscope := "zones"
    if IDtype == "account" {
        APIscope := "accounts"
    }
    uri := fmt.Sprintf("/%s/%s/logpush/jobs", APIscope, ID)
}
```
similarly, Logpush jobs could get scope-specific functions:
```go
func (api *API) CreateAccountLogpushJob(ctx context.Context, accountID string, job LogpushJob) (*LogpushJob, error) {
    return CreateLogpushJob(ctx, accountID, job, "account")
}
func (api *API) CreateZoneLogpushJob(ctx context.Context, zoneID string, job LogpushJob) (*LogpushJob, error) {
    return CreateLogpushJob(ctx, zoneID, job, "zone")
}
```

### References

https://github.com/cloudflare/terraform-provider-cloudflare/issues/1311

### PR DESCRIPTION
**Pull request by tjstansell**
## Description

This closes #750 and changes the logpush functions to use zone and account-specific function names that each take either the zoneID or accountID as an argument.  The previous names remain for compatibility and a deprecation message has been added to their log comment, pointing folks to the new zone-specific names.

## Has your change been tested?

No additional tests have been created.  I've renamed the existing logpush functions and marked the old names as deprecated in the log comments.  I still need to update the tests to test the account-level additions.

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.

### CODE REVIEW

**Comment by tjstansell** [comment]
I'm very new to `go` and never worked with this repo before, so if you have suggestions on how to write some additional tests for the new account-level endpoints, that'd be great.

**Comment by jacobbednarz** [comment]
at first glance, this looks like the right approach, kudos. when I’m in front of a computer next, I’ll double check this in finger detail before we merge it.

File: logpush.go [hunk 1]
```diff
@@ -92,11 +92,31 @@ type LogpushDestinationExistsRequest struct {
...
+// API reference: https://api.cloudflare.com/#logpush-jobs-create-logpush-job
+func (api *API) CreateZoneLogpushJob(ctx context.Context, zoneID string, job LogpushJob) (*LogpushJob, error) {
+	return api.createLogpushJob(ctx, ZoneRouteRoot, zoneID, job)
+}
+
+// CreateLogpushJob creates a new zone-level Logpush Job.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-create-logpush-job
+//
+// Deprecated: Renamed to CreateZoneLogpushJob.
```
Comment (jacobbednarz) [inline logpush.go:113]:
```suggestion
// Deprecated: Use `CreateZoneLogpushJob` or `CreateAccountLogpushJob` depending 
// on the desired resource to target.
```

File: logpush.go [hunk 2]
```diff
@@ -109,11 +129,31 @@ func (api *API) CreateLogpushJob(ctx context.Context, zoneID string, job Logpush
...
 // API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs
+func (api *API) ListZoneLogpushJobs(ctx context.Context, zoneID string) ([]LogpushJob, error) {
+	return api.listLogpushJobs(ctx, ZoneRouteRoot, zoneID)
+}
+
+// LogpushJobs returns all zone-level Logpush Jobs for all datasets.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs
+//
+// Deprecated: Renamed to ListZoneLogpushJobs.
```
Comment (jacobbednarz) [inline logpush.go:150]:
```suggestion
// Deprecated: Use `ListZoneLogpushJobs` or `ListAccountLogpushJobs`
// depending on the desired resource to target.
```

File: logpush.go [hunk 3]
```diff
@@ -143,11 +203,33 @@ func (api *API) LogpushJobsForDataset(ctx context.Context, zoneID, dataset strin
...
+// Zone Fields: https://developers.cloudflare.com/logs/reference/log-fields/zone
+func (api *API) GetZoneLogpushFields(ctx context.Context, zoneID, dataset string) (LogpushFields, error) {
+	return api.getLogpushFields(ctx, ZoneRouteRoot, zoneID, dataset)
+}
+
 // LogpushFields returns fields for a given dataset.
 //
 // API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs
+//
+// Deprecated: Renamed to GetZoneLogpushFields.
```
Comment (jacobbednarz) [inline logpush.go:226]:
```suggestion
// Deprecated: Use `GetZoneLogpushFields` or `GetAccountLogpushFields` 
// depending on the desired resource to target.
```

File: logpush.go [hunk 4]
```diff
@@ -160,11 +242,31 @@ func (api *API) LogpushFields(ctx context.Context, zoneID, dataset string) (Logp
...
+// API reference: https://api.cloudflare.com/#logpush-jobs-logpush-job-details
+func (api *API) GetZoneLogpushJob(ctx context.Context, zoneID string, jobID int) (LogpushJob, error) {
+	return api.getLogpushJob(ctx, ZoneRouteRoot, zoneID, jobID)
+}
+
 // LogpushJob fetches detail about one Logpush Job for a zone.
 //
 // API reference: https://api.cloudflare.com/#logpush-jobs-logpush-job-details
+//
+// Deprecated: Renamed to GetZoneLogpushJob.
```
Comment (jacobbednarz) [inline logpush.go:263]:
```suggestion
// Deprecated: Use `GetZoneLogpushJob` or `GetAccountLogpushJob` 
// depending on the desired resource to target.
```

File: logpush.go [hunk 5]
```diff
@@ -177,11 +279,31 @@ func (api *API) LogpushJob(ctx context.Context, zoneID string, jobID int) (Logpu
...
+// API reference: https://api.cloudflare.com/#logpush-jobs-update-logpush-job
+func (api *API) UpdateZoneLogpushJob(ctx context.Context, zoneID string, jobID int, job LogpushJob) error {
+	return api.updateLogpushJob(ctx, ZoneRouteRoot, zoneID, jobID, job)
+}
+
 // UpdateLogpushJob lets you update a Logpush Job.
 //
 // API reference: https://api.cloudflare.com/#logpush-jobs-update-logpush-job
+//
+// Deprecated: Renamed to UpdateZoneLogpushJob.
```
Comment (jacobbednarz) [inline logpush.go:300]:
```suggestion
// Deprecated: Use `UpdateZoneLogpushJob` or `UpdateAccountLogpushJob` 
// depending on the desired resource to target.
```

File: logpush.go [hunk 6]
```diff
@@ -194,11 +316,31 @@ func (api *API) UpdateLogpushJob(ctx context.Context, zoneID string, jobID int,
...
+// API reference: https://api.cloudflare.com/#logpush-jobs-delete-logpush-job
+func (api *API) DeleteZoneLogpushJob(ctx context.Context, zoneID string, jobID int) error {
+	return api.deleteLogpushJob(ctx, ZoneRouteRoot, zoneID, jobID)
+}
+
 // DeleteLogpushJob deletes a Logpush Job for a zone.
 //
 // API reference: https://api.cloudflare.com/#logpush-jobs-delete-logpush-job
+//
+// Deprecated: Renamed to DeleteZoneLogpushJob.
```
Comment (jacobbednarz) [inline logpush.go:337]:
```suggestion
// Deprecated: Use `DeleteZoneLogpushJob` or `DeleteAccountLogpushJob` 
// depending on the desired resource to target.
```

File: logpush.go [hunk 7]
```diff
@@ -211,11 +353,31 @@ func (api *API) DeleteLogpushJob(ctx context.Context, zoneID string, jobID int)
...
+// API reference: https://api.cloudflare.com/#logpush-jobs-get-ownership-challenge
+func (api *API) GetZoneLogpushOwnershipChallenge(ctx context.Context, zoneID, destinationConf string) (*LogpushGetOwnershipChallenge, error) {
+	return api.getLogpushOwnershipChallenge(ctx, ZoneRouteRoot, zoneID, destinationConf)
+}
+
 // GetLogpushOwnershipChallenge returns ownership challenge.
 //
 // API reference: https://api.cloudflare.com/#logpush-jobs-get-ownership-challenge
+//
+// Deprecated: Renamed to GetZoneLogpushOwnershipChallenge.
```
Comment (jacobbednarz) [inline logpush.go:374]:
```suggestion
// Deprecated: Use `GetZoneLogpushOwnershipChallenge` or
// `GetAccountLogpushOwnershipChallenge` depending on the 
// desired resource to target.
```

File: logpush.go [hunk 8]
```diff
@@ -235,11 +397,31 @@ func (api *API) GetLogpushOwnershipChallenge(ctx context.Context, zoneID, destin
...
 // API reference: https://api.cloudflare.com/#logpush-jobs-validate-ownership-challenge
+func (api *API) ValidateZoneLogpushOwnershipChallenge(ctx context.Context, zoneID, destinationConf, ownershipChallenge string) (bool, error) {
+	return api.validateLogpushOwnershipChallenge(ctx, ZoneRouteRoot, zoneID, destinationConf, ownershipChallenge)
+}
+
+// ValidateLogpushOwnershipChallenge returns zone-level ownership challenge validation result.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-validate-ownership-challenge
+//
+// Deprecated: Renamed to ValidateZoneLogpushOwnershipChallenge.
```
Comment (jacobbednarz) [inline logpush.go:418]:
```suggestion
// Deprecated: Use `ValidateZoneLogpushOwnershipChallenge` or
// `ValidateAccountLogpushOwnershipChallenge` depending on the 
// desired resource to target.
```

File: logpush.go [hunk 9]
```diff
@@ -255,11 +437,31 @@ func (api *API) ValidateLogpushOwnershipChallenge(ctx context.Context, zoneID, d
...
+// API reference: https://api.cloudflare.com/#logpush-jobs-check-destination-exists
+func (api *API) CheckZoneLogpushDestinationExists(ctx context.Context, zoneID, destinationConf string) (bool, error) {
+	return api.checkLogpushDestinationExists(ctx, ZoneRouteRoot, zoneID, destinationConf)
+}
+
+// CheckLogpushDestinationExists returns zone-level destination exists check result.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-check-destination-exists
+//
+// Deprecated: Renamed to CheckZoneLogpushDestinationExists.
```
Comment (jacobbednarz) [inline logpush.go:458]:
```suggestion
// Deprecated: Use `CheckZoneLogpushDestinationExists` or
// `CheckAccountLogpushDestinationExists` depending 
// on the desired resource to target.
```

File: logpush.go [hunk 10]
```diff
@@ -126,11 +166,31 @@ func (api *API) LogpushJobs(ctx context.Context, zoneID string) ([]LogpushJob, e
...
+// API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs-for-a-dataset
+func (api *API) ListZoneLogpushJobsForDataset(ctx context.Context, zoneID, dataset string) ([]LogpushJob, error) {
+	return api.listLogpushJobsForDataset(ctx, ZoneRouteRoot, zoneID, dataset)
+}
+
+// LogpushJobsForDataset returns all zone-level Logpush Jobs for a dataset.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs-for-a-dataset
+//
+// Deprecated: Renamed to ListZoneLogpushJobsForDataset.
```
Comment (jacobbednarz) [inline logpush.go:187]:
```suggestion
// Deprecated: Use `ListZoneLogpushJobsForDataset` or 
// `ListAccountLogpushJobsForDataset` depending on the desired resource
// to target.
```

File: logpush.go [hunk 11]
```diff
@@ -143,11 +203,33 @@ func (api *API) LogpushJobsForDataset(ctx context.Context, zoneID, dataset strin
 	return r.Result, nil
 }
 
+// GetAccountLogpushFields returns fields for a given account-level dataset.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs
+// Account Fields: https://developers.cloudflare.com/logs/reference/log-fields/account
```
Comment (jacobbednarz) [inline logpush.go:209]:
```suggestion
// GetAccountLogpushFields returns fields for a given account-level dataset.
//
// Account fields documentation: https://developers.cloudflare.com/logs/reference/log-fields/account
//
// API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs
```

File: logpush.go [hunk 12]
```diff
@@ -143,11 +203,33 @@ func (api *API) LogpushJobsForDataset(ctx context.Context, zoneID, dataset strin
...
+func (api *API) GetAccountLogpushFields(ctx context.Context, accountID, dataset string) (LogpushFields, error) {
+	return api.getLogpushFields(ctx, AccountRouteRoot, accountID, dataset)
+}
+
+// GetZoneLogpushFields returns fields for a given zone-level dataset.
+//
+// API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs
+// Zone Fields: https://developers.cloudflare.com/logs/reference/log-fields/zone
```
Comment (jacobbednarz) [inline logpush.go:217]:
```suggestion
// GetZoneLogpushFields returns fields for a given zone-level dataset.
//
// Zone fields documentation: https://developers.cloudflare.com/logs/reference/log-fields/zone
//
// API reference: https://api.cloudflare.com/#logpush-jobs-list-logpush-jobs
```

**Review by jacobbednarz — APPROVED** [review]
LGTM! just nitpicky comment formatting to make it nicer in editors.

solid effort here!

**Pull request merged**